dev/create-release/release-build.sh

```diff
@@ -116,14 +116,17 @@ SCALA_2_11_PROFILES=
 if [[ $SPARK_VERSION > "2.3" ]]; then
   BASE_PROFILES="$BASE_PROFILES -Pkubernetes"
   if [[ $SPARK_VERSION < "3.0." ]]; then
-    SCALA_2_11_PROFILES="-Pkafka-0-8"
+    SCALA_2_11_PROFILES="-Pkafka-0-8 -Pflume"
   fi
```

Comment:
Hi, @vanzin , @dbtsai , @srowen , @cloud-fan .

For `branch-2.4` release, I thought we are supposed to use `dev/create-release/release-build.sh` in `branch-2.4`.

`branch-2.4` already has `-Pflume` [here](https://github.com/apache/spark/blob/branch-2.4/dev/create-release/release-build.sh#L117-L118). Is this PR aiming to use `master` branch code to do release `2.4.x`?
```bash
if [[ $SPARK_VERSION > "2.3" ]]; then
  BASE_PROFILES="$BASE_PROFILES -Pkubernetes -Pflume"
  SCALA_2_11_PROFILES="-Pkafka-0-8"
else
```

Reply:
In the past I've avoided using the release scripts from the branches for a bunch of reasons. But mainly:

- if you find a problem with the release script in the middle of building an RC, it's awkward to change the script in the branch without creating a new RC.
- as master moves forward, it becomes tricky to backport fixes from the master scripts to other branches

So I've always used the release scripts from master, and I think it's much simpler for everybody if they also do that.

(Note that even though the 2.4 script has the flume profile enabled, 2.4.0 was shipped without a flume artifact published to maven.)

Reply:
I had thought the same, but I think the practice is to let the main script create releases for prior branches too. I'd defer to release managers for policy on this. We ought to be able to drop support in the script for EOL releases. I did that a while ago in order to keep it at all sane after adding scala 2.12 support.

Reply:
Thanks, @vanzin and @srowen . It would be great if we explicitly guide to use the `master` branch's release script always at https://spark.apache.org/release-process.html .